core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -105,14 +95,14 @@ public void delete(TableIdentifier tableIdentifier, Table table, RemovalCause ca
   }
 
   private Cache<TableIdentifier, Table> createTableCache(Ticker ticker) {
-    Caffeine<TableIdentifier, Table> cacheBuilder = Caffeine
+    Caffeine<Object, Object> cacheBuilder = Caffeine
         .newBuilder()
-        .softValues()
-        .removalListener(identLoggingRemovalListener);
+        .softValues();
 
     if (expirationIntervalMillis > 0) {
       return cacheBuilder
-          .writer(new CachingCatalog.MetadataTableInvalidatingCacheWriter())
+          .removalListener(new MetadataTableInvalidatingRemovalListener())
+          .executor(Runnable::run)
```

Comment:
Can you put a comment inline next to this so we don't forget why we did this in the future?

```
      return cacheBuilder
          .removalListener(new MetadataTableInvalidatingRemovalListener())
          .executor(Runnable::run)  // Makes the callbacks to removal listener synchronous
```

Reply:
done :D